### DISCUSSION TITLE
Migration: Triggers

### DISCUSSION THREAD

**Discussion by johnshift** [supabase/supabase#5692] [Questions] [answered]
First I created a `local instance` by following the steps provided [here](https://github.com/supabase/cli/tree/main/examples/tour).
Then I created a function along with a trigger that populates the `users` table after signup:
<details>
<summary>click to show sql</summary>

```sql
  create or replace function public.handle_after_signup()
    returns trigger
    language plpgsql
    security definer set search_path = public as
  $$
  begin
    -- populate user table
    insert into public.users (id, email, username, name)
    values (
      new.id,
      new.email,
      new.raw_user_meta_data ->> 'username',
      new.raw_user_meta_data ->> 'name'
    );
	
    -- clear user metadata on auth.users
    update auth.users set raw_user_meta_data = '{}' where email = new.email;
    return new;
  end;
  $$;

create trigger after_signup_trigger
after insert on auth.users
  for each row execute procedure public.handle_after_signup();
```
</details>

Everything's fine when I ran it in `SQL Editor`  at my `local instance`.
After following the migration steps [here](https://github.com/supabase/cli/tree/main/examples/tour), the `trigger` is not included in the `remote instance`.
I am wondering if this is really the case, or is it part of the `not production ready` notice?
Thank you in advance!

**Comment by GaryAustin1** [comment, marked as answer]
Probably this based on your trigger being on auth:
https://github.com/supabase/cli/issues/120
The not ready for production on triggers is just the UI.  Triggers are part of postgres itself.

**Reply by johnshift** [reply]
Thank you! [#120](https://github.com/supabase/cli/issues/120) is exactly what I was looking for.
